src/main/java/com/couchbase/lite/replicator/PusherInternal.java

```diff
@@ -295,12 +295,25 @@ public void beginReplicating() {
     @InterfaceAudience.Private
     public void changed(Database.ChangeEvent event) {
         final List<DocumentChange> changes = event.getChanges();
-        supportExecutor.submit(new Runnable() {
-            @Override
-            public void run() {
-                submitRevisions(changes);
-            }
-        });
+
+        // NOTE: REST API client can send POST/PUT /{db}/{doc} request too faster than
+        // push replicator can handle. If request comes from listener thread, submitting revision
+        // to batcher synchronusly. Otherwise, asynchronously.
+        // ref: https://github.com/couchbase/couchbase-lite-java-core/issues/1481
+        if (calledFromListener()) {
+            submitRevisions(changes);
+        } else {
+            supportExecutor.submit(new Runnable() {
+                @Override
+                public void run() {
+                    submitRevisions(changes);
+                }
+            });
+        }
+    }
+
```

Comment:
I have a couple of questions.
 1. If I remember correctly, the submitRevisions() is called in the supportExecutor to fix an issue, probably related to threading issues. Is it ok to call submitRevisions outside the supportExecutor?
 2. From the note above that REST API client can send POST/PUT /{db}/{doc} request too faster than push replicator can handle, isn't the pause logic in push replicator supposed to prevent this issue? Also Can't the native android application insert the documents as fast as the REST client?